sui-execution/latest/sui-move-natives/src/accumulator.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    object_runtime::{MoveAccumulatorAction, MoveAccumulatorValue, ObjectRuntime},
+    NativesCostTable,
+};
+use move_binary_format::errors::PartialVMResult;
+use move_core_types::account_address::AccountAddress;
+use move_vm_runtime::{native_charge_gas_early_exit, native_functions::NativeContext};
+use move_vm_types::{
+    loaded_data::runtime_types::Type, natives::function::NativeResult, values::Value,
+};
+use smallvec::smallvec;
+use std::collections::VecDeque;
+use sui_types::base_types::ObjectID;
+
+pub fn emit_deposit_event(
+    context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    emit_event(context, ty_args, args, MoveAccumulatorAction::Merge)
+}
+
+pub fn emit_withdraw_event(
+    context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    emit_event(context, ty_args, args, MoveAccumulatorAction::Split)
+}
+
+fn emit_event(
+    context: &mut NativeContext,
+    mut ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+    action: MoveAccumulatorAction,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(ty_args.len() == 1);
+    debug_assert!(args.len() == 3);
```

Comment:
nit, would be preferable to move these into each function I think... maybe... no strong opinions though. 
It is just not common to be doing this sort of shared behavior between natives in this way. In other words, we typically only do the Rust implementation items in these shared functions--not the processing of args/returns